internal/users/validation.go

```diff
@@ -0,0 +1,164 @@
+package users
+
+import (
+	"gopkg.in/go-playground/validator.v9"
+	"net/mail"
+	"regexp"
+	"strings"
+	"time"
+	"unicode"
+)
+
+func HasDigit(password string) bool {
+	for _, char := range password {
+		if unicode.IsDigit(char) {
+			return true
+		}
+	}
+	return false
+}
+
+func HasCommonWord(password string) bool {
+	commonWords := []string{"password", "12345678", "87654321", "qwerty123"}
+	for _, word := range commonWords {
+		if strings.Contains(password, word) {
+			return true
+		}
+	}
+	return false
+}
+
+func HasSequence(password string) bool {
+	sequences := []string{"123", "abc", "xyz"}
+	for _, sequence := range sequences {
+		if strings.Contains(password, sequence) {
+			return true
+		}
+	}
+	return false
+}
+func CheckPassword(fl validator.FieldLevel, v *UserValid) bool {
+	password := fl.Field().String()
+	if len(password) < 8 {
+		v.validErr["password"] += "short,"
+		return false
+	}
+	if len(password) > 100 {
+		v.validErr["password"] += "long,"
+		return false
+	}
+	hasUpperCase := false
+	hasSpecialChar := false
+	hasDigit := false
+	hasSequence := false
+	hasCommonWord := false
+
+	for _, char := range password {
+		if unicode.IsUpper(char) {
+			hasUpperCase = true
+		} else if !unicode.IsLetter(char) && !unicode.IsNumber(char) {
+			hasSpecialChar = true
+		}
+	}
+
+	if !HasDigit(password) == false {
+		hasDigit = true
+	}
+
+	if HasSequence(password) == false {
+		hasSequence = true
+	}
+
+	if HasCommonWord(password) == false {
+		hasCommonWord = true
+	}
+	if hasUpperCase == false {
+		v.validErr["password"] += "uppercase,"
+	}
+	if hasSpecialChar == false {
+		v.validErr["password"] += "special character,"
+	}
+	if hasDigit == false {
+		v.validErr["password"] += "digit,"
+	}
+	if hasSequence == false {
+		v.validErr["password"] += "sequence,"
+	}
+	if hasCommonWord == false {
+		v.validErr["password"] += "common word,"
+	}
+	if (hasUpperCase && hasSpecialChar && hasDigit && hasSequence && hasCommonWord) == false {
+		return false
+	}
+
+	return true
+}
+
+func CheckDataTime(fl validator.FieldLevel, v *UserValid) bool {
```

Comment:
![image](https://github.com/mo-pm888/twitter/assets/1732280/cb5471d6-79f4-4773-9669-62e472870780)

Reply:
also, you are validating date, not datetime